Added verbose flag to update.sh (issue #11574)

haabe opened the pull request:
## Standards checklist:

<!-- Fill with an x the ones that apply. Example: [x] -->

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- Sources settings from $HOME/.zshrc to read :omz:update verbose
- Sets verbose_mode to default if not set
- Dependant text rendering based on verbose_mode

## Other comments:
- .zshrc probably should have a mention of :omz:update verbose if implemented
- docs probably should have a mention of :omz:update verbose if implemented
...

carlosala commented on tools/upgrade.sh:
```diff
@@ -203,7 +214,10 @@ git checkout -q "$branch" -- || exit 1
 # Update Oh My Zsh
-printf "${BLUE}%s${RESET}\n" "Updating Oh My Zsh"
+# Avoid printing if silent
+if [[ "$verbose_mode" != silent ]]
```
missing `; then`

carlosala commented on tools/upgrade.sh:
```diff
@@ -218,21 +232,27 @@ if LANG= git pull --quiet --rebase $remote $branch; then
-    printf "${BLUE}%s \`${BOLD}%s${RESET}${BLUE}\`${RESET}\n" "You can see the changelog with" "omz changelog"
+    
+    # Skip printing if silent
+    if [[ "$verbose_mode" != silent ]]
```
missing `; then`

carlosala commented on tools/upgrade.sh:
```diff
@@ -218,21 +232,27 @@ if LANG= git pull --quiet --rebase $remote $branch; then
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "To keep up with the latest news and updates, follow us on Twitter:" "$(fmt_link @ohmyzsh https://twitter.com/ohmyzsh)"
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "Want to get involved in the community? Join our Discord:" "$(fmt_link "Discord server" https://discord.gg/ohmyzsh)"
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "Get your Oh My Zsh swag at:" "$(fmt_link "Planet Argon Shop" https://shop.planetargon.com/collections/oh-my-zsh)"
+  if [[ "$verbose_mode" = default]]
```
let's go for `==` and a space after default, and `; then`

carlosala commented on tools/upgrade.sh:
```diff
@@ -218,21 +229,27 @@ if LANG= git pull --quiet --rebase $remote $branch; then
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "To keep up with the latest news and updates, follow us on Twitter:" "$(fmt_link @ohmyzsh https://twitter.com/ohmyzsh)"
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "Want to get involved in the community? Join our Discord:" "$(fmt_link "Discord server" https://discord.gg/ohmyzsh)"
-  printf "${BLUE}${BOLD}%s %s${RESET}\n" "Get your Oh My Zsh swag at:" "$(fmt_link "Planet Argon Shop" https://shop.planetargon.com/collections/oh-my-zsh)"
+  if [[ "$verbose_mode" == default]]; then
```
A space missing after `default`

carlosala reviewed (changes requested):
I'll test a bit more and get back. In the meantime, could you also document this feature in main README?

haabe commented:
So I rewrote it to pass an argument together with --interactive in the cli - and fixed my impressively embarrassing `; then` bugs. I also added a few lines about the config in README.md.

carlosala commented:
Thanks for your contribution! I changed a few things that made sense for me, but the functionality is there! Thanks for your contribution!
